### DISCUSSION TITLE
magit slowdown with excessive use

### DISCUSSION THREAD

**Discussion by nkiesel** [magit/magit#4450] [Support] [answered]
Hi,

I love Magit and use it especially when dealing with complicated commits.  The only problem I have is that with every use, Magit gets slower and slower.  While the first diff shown for a commit - even if the diff contains changes for dozens of files - is fast (1-2 seconds), after a couple of such commits the "commit diff" takes up to a minute and Magit is frozen during that time.

The only solution I'm aware of is to restart Emacs,

My laptop of reasonably powerful (1 year old Dell, 1GB memory, Intel i-7, SSD), and other resource-intensive stuff (e.g. IntelliJ) continues to perform w/o much performance regressions.

Am I doing something wrong?

**Comment by alphapapa** [comment]
Why did you also open an issue (#4451) before waiting for an answer here?  Many thousands of people use Magit without this problem, so it's likely specific to your configuration and not a bug in Magit.

Now, you haven't given any information to help diagnose the problem, like OS, Emacs version, Magit version, git version, repo size, whether it happens in a clean Emacs config, etc.

(And does your 1-year-old i7 laptop really only have 1 GB of RAM?)

**Reply by nkiesel** [reply]
Sorry for the confusion.  I first created a "question" but then there was link saying " no answers, do you want to create a new issue?" and I clicked it.  Again, sorry for the noise.

And yes: my laptop has 32GB of memory, not just 1.  I am using

- Dell Latitude 5400
   - Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz
   - 32GB memory
   - 1 TB SSD
- Debian unstable 64bit (Debian GNU/Linux 11 (bullseye))
- Wayland from Debian unstable
- Linux 5.10.0-8-amd64 from Debian unstable
- Emacs 27.1 from Debian unstable
   - 195 installed packages
- Git 2.32.0 from Debian unstable
- Git repo:
   - 90000 commits
   - 27000 files
   - .git size 1.4GB
- Magit installed via "list-packages"
   - magit (20210721.1556)
   - magit-delta (20210104.1541)
   - magit-find-file (20150702.830)
   - magit-libgit (20210702.819)
   - magit-rbr (20181009.2016)
   - magit-svn  (20210426.2114)
   - magit-popup  (20200719.1015)
   - magit-section ( 20210702.822)

I understand that lot of the SW is "bleeding edge".  However, i do update regularly and I have seen the slowdown for months now.

Is there any way I can get metrics out of Magit or Emacs to help pinning down my issue?

**Reply by alphapapa** [reply]
Do note that Emacs 27.1 is not the latest release, but I wouldn't expect that to be a problem.

What I do notice is the size of that git repo: 27,000 files is a lot.  However, that shouldn't matter except for `git` operations which output a large number of lines--most operations on even large repos don't do that.  And besides, the problem you've described, a kind of gradual slowdown, is weird.

I also notice that you have some extension packages that aren't part of Magit itself; one of them, at least, is 6 years old.

So I'd recommend trying to reproduce the problem by starting with a clean Emacs configuration, installing only `magit`.  You could use [this script](https://github.com/alphapapa/emacs-sandbox.sh) to run that clean Emacs configuration, while still using your main config to do your normal work.

**Comment by nkiesel** [comment, marked as answer]
I will try this for a while and then report back here
